zschema/keys.py

```diff
@@ -30,6 +30,27 @@ def __cmp__(self, other):
             return cmp(int(self.port), int(other.port))
 
 
+# Create a dict with all the keys / values of rhs and lhs, where values
+# from lhs are used if a key is in both.
+def left_merge(lhs, rhs):
+    return dict(v for v in rhs.items() + lhs.items())
+
+
+# Factory for TypeFactorys (e.g. SubRecordType is a TypeFactory for
+# SubRecords).
+class TypeFactoryFactory():
```

Comment:
This should probably be a new style class, i.e., inherit object

Reply:
d'oh. Fixing.